content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
 backlog_filter= 'Week 2'
 +++
 
+## Learning objectives
 
+- Label and describe the syntactic features of a function declaration
+- Identify a call site
+- Differentiate between a function declaration and a function invocation
+- Propose a function definition to satisfy some problem requirements
+- Given a short program, determine which variable names are _bad_ choices
+- Give examples of when the `undefined` data type may occur in a JavaScript program
+- Construct simple expressions to represent a set of operations
+- Define a _parameter_
+- Identify the return value a call site given its function declaration
+- Identify the value assigned to a parameter when a function is invoked
+- Identify the type of a return value given some call site
+- Given a short program, state the number of times a given function is called
+- Differentiate between parameters and arguments
+- Invoke a given function with appropriate arguments to produce some target output
+- Describe how to log the output of a function
+- Analyse each line in a given program and state what the line does when the program executes
+- Explain the difference between returning a value and logging a value
+- Assess and explain whether a function can reference a given variable or not
+- Explain what is meant by the term "implementation opacity"
+- Explain how to reuse some code to perform a specific task
+- Propose clear and meaningful parameter names for a function
```

Comment:
I don't think we've talked about this enough yet.

Reply:
I think we should address this learning objective in depth in one of the coursework issues